TypeScript/test/mocha/gilded-rose.spec.ts

```diff
@@ -3,74 +3,145 @@ import { Item, GildedRose } from '@/gilded-rose';
-    expect(items[0].sellIn).to.equal(30);
-  });
-  it('should increase quality for backstage passes', () => {
-    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 20, 30)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(31);
-  });
-  it('should increase quality for backstage passes by 2 when 10 days are left', () => {
-    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 10, 30)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(32);
-  });
-  it('should increase quality for backstage passes by 3 when 5 days are left', () => {
-    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 5, 40)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(43);
-  });
-  it('should set quality for backstage passes to 0 after the date', () => {
-    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 0, 30)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(0);
+
+  describe('Aged Brie', () => {
+    it('should increase quality for aged brie', () => {
+      //given
+      const gildedRose = new GildedRose([new Item('Aged Brie', 2, 25)]);
+      //when
+      const items = gildedRose.updateQuality();
+      //then
+      expect(items[0].quality).to.equal(26);
+    });
+
+    it('should increase quality by 2 for aged brie past sellIn', () => {
+      //given
+      const gildedRose = new GildedRose([new Item('Aged Brie', 0, 30)]);
+      //when
+      const items = gildedRose.updateQuality();
+      //then
+      expect(items[0].quality).to.equal(32);
+    });
   });
-  it('should increase quality by 2 for aged brie past sellin', () => {
-    const gildedRose = new GildedRose([new Item('Aged Brie', 0, 30)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(32);
+
+  describe('Sulfuras', () => {
+    it('should not decrease quality for Sulfuras', () => {
+      //given
+      const gildedRose = new GildedRose([new Item('Sulfuras, Hand of Ragnaros', 30, 50)]);
+      //when
+      const items = gildedRose.updateQuality();
+      //then
+      expect(items[0].quality).to.equal(50);
+    });
+
+    it('should not decrease sellIn for Sulfuras', () => {
+      //given
+      const gildedRose = new GildedRose([new Item('Sulfuras, Hand of Ragnaros', 30, 50)]);
+      //when
+      const items = gildedRose.updateQuality();
+      //then
+      expect(items[0].sellIn).to.equal(30);
+    });
   });
-  it('should decrease Conjured quality by 2', () => {
-    const gildedRose = new GildedRose([new Item('Conjured', 10, 30)]);
-    const items = gildedRose.updateQuality();
-    expect(items[0].quality).to.equal(28);
+
+  describe('Backstage passes', () => {
+    it('should increase quality for backstage passes', () => {
```

Comment:
now that we use nested describes, we don't need the item types in the names of the tests anymore